src/MinimalDelegation.sol

```diff
@@ -191,10 +194,18 @@ contract MinimalDelegation is
     {
         (bytes32 keyHash, bytes memory signature, bytes memory hookData) =
             abi.decode(wrappedSignature, (bytes32, bytes, bytes));
-        bytes32 digest = _hashTypedData(data.hashWithWrappedType());
 
         Key memory key = getKey(keyHash);
-        bool isValid = key.verify(digest, signature);
+
+        // Must be branched because we do abi decoding in memory which will throw since the ecnoding schemes are different
```

Comment:
Flag here also that decoding in memory makes the implementation a little awkaward. Comment here says more